### PR TITLE
[neulion] Updated the ufctv plugin to make it useable for other domains

### PR DESCRIPTION
**Pull request by back-to**
#### [neulion] renamed ufctv.py to neulion.py 

---

#### [neulion] Updated the ufctv plugin to make it useable for other domains

- Fixed KeyError if there is no 'type' in the result data

- **--ufctv-username** and **--ufctv-password** can still be used,
  but the new command is **--neulion-username** **--neulion-password**

it should work for
```
https://fanpass.co.nz - Tested
https://watch.nba.com - only free Video Tested
https://watch.rugbypass.com - Not Tested
https://www.elevensports.be - only free Video Tested
https://www.elevensports.lu - only free Video Tested
https://www.elevensports.pl - only free Video Tested
https://www.elevensports.sg - Tested
https://www.elevensports.tw - Tested
https://www.tennischanneleverywhere.com - Tested
https://www.ufc.tv - old plugin nothing changed
```

Fixed streamlink/streamlink#1037
Fixed streamlink/streamlink#1130
Fixed streamlink/streamlink#1185

- Added regex tests

---

#### [neulion] Replace `'` with `"` in js_to_json, show Login required for 400 error in _get_stream_url

---

_**Known issue:**_
- the **ufctv.py** file might not be removed for some windows users streamlink/streamlink#1223

### CODE REVIEW

File: src/streamlink/plugins/neulion.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+from streamlink.plugin import Plugin, PluginOptions
+from streamlink.plugin.api import http
+from streamlink.plugin.api import useragents
+from streamlink.plugin.api import validate
+from streamlink.stream import HLSStream
+from streamlink.utils import parse_json
+
+
+def js_to_json(data):
```
Comment (back-to) [inline src/streamlink/plugins/neulion.py]: just saw that this **js_to_json** can't handle some data with `:` or `,` in names were well.

it needs an update 🌊

**Comment by karlo2105** [comment]
@back-to 
Another domain using Neulion : `http://sportspass.ca/live/` ;)

**Comment by back-to** [comment]
it's not a neulion domain.

**Comment by gravyboat** [comment]
Thanks @back-to!

**Pull request merged**